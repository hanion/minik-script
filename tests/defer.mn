// defer.mn

print("basic defer execution");
{
	var := "";
	{
		defer var = "deferred";
		var = "should not print this";
	}
	print(var);
}

print("multiple defers execute in reverse order");
{
	defer print("1");
	defer print("2");
	defer print("3");
}

print("nested defer statements");
{
	defer print("outer 1");
	{
		defer print("inner 1");
		defer print("inner 2");
	}
	defer print("outer 2");
}

print("defer in loops");
for i := 0; i < 3; ++i {
	defer print("deferred in loop", i);
	print("loop iteration", i);
}

print("defer in loops with break");
for i := 0; i < 3; ++i {
	defer print("deferred in loop", i);
	if i == 1 {
		break;
	}
}

print("defer in loops with continue");
for i := 0; i < 3; ++i {
	defer print("deferred in loop", i);
	if i == 1 {
		continue;
	}
}
