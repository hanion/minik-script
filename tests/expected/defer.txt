basic defer execution
deferred
multiple defers execute in reverse order
3
2
1
nested defer statements
inner 2
inner 1
outer 2
outer 1
defer in loops
loop iteration
0.000000
deferred in loop
0.000000
loop iteration
1.000000
deferred in loop
1.000000
loop iteration
2.000000
deferred in loop
2.000000
defer in loops with break
deferred in loop
0.000000
deferred in loop
1.000000
defer in loops with continue
deferred in loop
0.000000
deferred in loop
1.000000
deferred in loop
2.000000
